### ISSUE TITLE
No handlers could be found for logger "trollius"

### ISSUE THREAD

**Issue by RDaneelOlivav** [jpieper/pygazebo#28]
Got this message when subscribing to a topic.
When activated logger, gives error:
ERROR:trollius:Exception in callback <lambda>(<Future ...00\xf0?'>) at /usr/local/lib/python2.7/dist-packages/pygazebo-3.0.0_2014.1-py2.7.egg/pygazebo/pygazebo.py:198
handle: <Handle <lambda>(<Future ...00\xf0?'>) at /usr/local/lib/python2.7/dist-packages/pygazebo-3.0.0_2014.1-py2.7.egg/pygazebo/pygazebo.py:198>
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/trollius-1.0.4-py2.7.egg/trollius/events.py", line 151, in _run
    self._callback(*self._args)
  File "/usr/local/lib/python2.7/dist-packages/pygazebo-3.0.0_2014.1-py2.7.egg/pygazebo/pygazebo.py", line 198, in <lambda>
    lambda future: self._handle_read(future, connection))
  File "/usr/local/lib/python2.7/dist-packages/pygazebo-3.0.0_2014.1-py2.7.egg/pygazebo/pygazebo.py", line 209, in _handle_read
    self._connect3(connection)
TypeError: _connect3() takes exactly 3 arguments (2 given)

The code is:
"""
import trollius
from trollius import From
import pygazebo
import pygazebo.msg.response_pb2
import logging
logging.basicConfig()

RESPONSE_TOPIC_NAME = '/gazebo/default/response'
RESPONSE_TOPIC_MSG_TYPE = 'gazebo.msgs.Response'
# SUBSCRIBER

def callback(data):
    response = pygazebo.msg.response_pb2.Response()
    if not data:
        print "No Data..."
    else:
        response.ParseFromString(data)
        print "Response ==>", response.response
        print "Response ==>", response.request

@trollius.coroutine
def listen_topics_loop():

```
manager = yield From(pygazebo.connect())
subscriber = manager.subscribe( RESPONSE_TOPIC_NAME,
                                RESPONSE_TOPIC_MSG_TYPE,
                                callback)
while True:
    #print "Listening"
    yield From(trollius.sleep(2.0))
```

def request_and_listen():
    loop = trollius.get_event_loop()
    loop.run_until_complete(listen_topics_loop())

if **name** == "**main**":
    request_and_listen()

"""

**Comment by RDaneelOlivav** [comment]
The error take you to here "/usr/local/lib/python2.7/dist-packages/pygazebo-3.0.0_2014.1-py2.7.egg/pygazebo/pygazebo.py"

Where we see something that it dazzels me a bit:

```
def _connect3(self, future, connection):
    future.result()  # check for error

    if not self._connection_future.done():
        self._connection_future.set_result(None)
    future = connection.read_raw()
    future.add_done_callback(
        lambda future: self._handle_read(future, connection))

def _handle_read(self, future, connection):
    data = future.result()
    if data is None:
        self._connections.remove(connection)
        if len(self._connections) == 0:
            self._connection_future = asyncio.Future()
        return

    self.callback(data)
    self._connect3(connection)
```

Shouldnt _connect3 have 2 inputs (connection and FUTURE) ?

Can any one clarify it a bit?

**Comment by jpieper** [comment]
As mentioned in the other ticket, this is fixed on develop as of d9137078b87d41cea2b350c08a0c229f6b4dac72

I will try to put out a release soonish, but in the meantime, feel free to check out the "develop" branch and use it directly.

**Comment by RDaneelOlivav** [comment]
Thanks ;). Solved it then ;)
